apisix/plugins/openwhisk.lua

```diff
@@ -0,0 +1,121 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local core              = require("apisix.core")
+local http              = require("resty.http")
+local ngx_encode_base64 = ngx.encode_base64
+local tostring          = tostring
+
+local schema = {
+    type = "object",
+    properties = {
+        api_host = {type = "string"},
+        ssl_verify = {
+            type = "boolean",
+            default = true,
+        },
+        service_token = {type = "string"},
+        namespace = {type = "string"},
+        action = {type = "string"},
+        result = {
+            type = "boolean",
+            default = true,
+        },
+        timeout = {
+            type = "integer",
+            minimum = 1000,
+            maximum = 60000,
```

Comment:
Please use the same timeout setting in azure-functions.

Reply:
It may not be able to complete the request within a maximum of 3000ms due to the need to download the runtime images or pull up a new service container. In addition, this timeout time acts on both HTTP requests from APISIX to OpenWhisk traffic entry and OpenWhisk internal action execution time consumption, if set to a lower value, it will cause all requests to fail.

BTW, the OpenWhisk‘s timeout support 1ms minimum, the range is 1ms - 60000ms. This is defined by OpenWhisk.

Reply:
Can you add your discovery as a comment?

Reply:
Of course, I will add it to documents.